### PR TITLE
Suggester2

### PR DESCRIPTION
**Pull request by jieluo321**
refactoring of suggester index

### CODE REVIEW

File: indexer/src/test/java/org/uniprot/store/indexer/uniprotkb/processor/UniProtEntryConverterTest.java [hunk 1]
```diff
@@ -367,14 +370,14 @@ void testConvertFullQ9EPI6Entry() throws Exception {
...
-        assertEquals(13, doc.subcellLocationTerm.size());
+        assertEquals(26, doc.subcellLocationTerm.size());
         assertTrue(doc.subcellLocationTerm.contains("Nucleus envelope"));
         assertEquals(0, doc.subcellLocationTermEv.size());
         assertEquals(1, doc.subcellLocationNote.size());
         assertEquals(2, doc.subcellLocationNoteEv.size());
         assertTrue(doc.subcellLocationNoteEv.contains("ECO_0000250"));
         assertTrue(doc.subcellLocationNoteEv.contains("manual"));
-        checkSuggestionsContain(SuggestDictionary.SUBCELL, doc.subcellLocationTerm, true);
+    //    checkSuggestionsContain(SuggestDictionary.SUBCELL, doc.subcellLocationTerm, true);
```
Comment (eddturner) [inline indexer/src/test/java/org/uniprot/store/indexer/uniprotkb/processor/UniProtEntryConverterTest.java]: can we uncomment this line to make sure the subcell values are tested?

Comment (jieluo321) [reply indexer/src/test/java/org/uniprot/store/indexer/uniprotkb/processor/UniProtEntryConverterTest.java]: the reason i comment out is we will refactor UniProtEntryConverter. The tests will more likely be rewritten

**Comment by LeonardoGonzales** [comment]
Hi Jie,

Do you know why it is failing? As you made the changes it is easier for you
to fix it.

Anyone else would need to spend time investigating this.

Best Regards

Leo

**Comment by jieluo321** [comment]
it is the size. subcellLocationTerm now contains value and id. Previously we only store value not id in subcellLocationTerm. I can re-filter id in the test so the  checkSuggestionsContain will work

**Comment by jieluo321** [comment]
i fixed and uncomment the line

**Pull request merged**